PubSub/src/Subscription.php

```diff
@@ -722,10 +723,15 @@ public function acknowledgeBatch(array $messages, array $options = [])
     {
         $this->validateBatch($messages, Message::class);
 
-        $this->connection->acknowledge($options + [
-            'subscription' => $this->name,
-            'ackIds' => $this->getMessageAckIds($messages)
-        ]);
+        // the rpc may throw errors for a sub with EOD enabled
```

Comment:
It looks like we have EOD configuration exposed through the GAPICs, but we haven't bubbled it up to the veneer yet. It looks like we should open another PR to surface it.

Reply:
This should be done :)